File: retropreference/src/main/java/com/twiceyuan/retropreference/typeHandler/StringSetHandler.java
package com.twiceyuan.retropreference.typeHandler;

import android.content.SharedPreferences;

import java.util.Set;

/**
 * Created by twiceYuan on 20/01/2017.
 *
 * StringSet handler
 */
class StringSetHandler extends BaseTypeHandler<Set<String>> {

    StringSetHandler(SharedPreferences preferences) {
        super(preferences);
    }

    @Override
    public void setValue(String key, Set<String> value) {
        mPreferences.edit().putStringSet(key, value).apply();
    }

    @Override
    public Set<String> get(String key, Set<String> defaultValue) {
        return mPreferences.getStringSet(key, defaultValue);
    }
}

File: retropreference/src/main/java/com/twiceyuan/retropreference/typeHandler/StringSetHandler.kt
package com.twiceyuan.retropreference.typeHandler

import android.annotation.TargetApi
import android.content.SharedPreferences
import android.os.Build

/**
 * Created by twiceYuan on 20/01/2017.

 * StringSet handler
 */
internal class StringSetHandler(preferences: SharedPreferences) : BaseTypeHandler<Set<String>>(preferences) {

    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
    override fun setValue(key: String, value: Set<String>?) {
        mPreferences.edit().putStringSet(key, value).apply()
    }

    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
    override fun get(key: String, defaultValue: Set<String>?): Set<String>? {
        return mPreferences.getStringSet(key, defaultValue)
    }
}
